index.bs

```diff
@@ -226,14 +323,17 @@ Their [=deserialization steps=], given |serialized| and |value| are:
      of a directory.
 
   : |handle| . {{FileSystemHandle/name}}
-  :: Returns the [=file system entry/name=] of the entry represented by |handle|.
+  :: Returns the last path component of |handle|'s
+     [=FileSystemHandle/locator=]'s [=file system locator/path=].
 </div>
 
 The <dfn attribute for=FileSystemHandle>kind</dfn> getter steps are to return
-{{FileSystemHandleKind/"file"}} if [=this=] is a [=file entry=]; otherwise
-{{FileSystemHandleKind/"directory"}}.
+[=this=]'s [=FileSystemHandle/locator=]'s [=file system locator/kind=].
 
-The <dfn attribute for=FileSystemHandle>name</dfn> getter steps are to return [=this=]'s [=file system entry/name=].
+The <dfn attribute for=FileSystemHandle>name</dfn> getter steps are to return
+the empty [=string=] if [=this=] is the root directory of the
+[=origin private file system=], otherwise the last [=list/item=] (a [=string=])
```

Comment:
```suggestion
[=origin private file system=]; otherwise the last [=list/item=] (a [=string=])
```

Reply:
I think "origin private file system"-ness needs to be stored on _this_ in a field somewhere or equivalent. This reads a little funky.

Reply:
Hmm, yeah it's somewhat implied by the "file system root", but I agree we could make it explicit. It would be nice to give each locator an optional "storage bucket", and if a bucket is set then the object must be from an OPFS (since non-OPFS files live outside the browser, and therefore the bucket system, entirely). But storage buckets aren't yet standardized, which is why I've only mentioned them in non-normative text for now

I'm happy to add a boolean field for now, but there are some other OPFS-related changes (e.g. #92 and https://github.com/whatwg/fs/pull/9/files#diff-5e793325cd2bfc452e268a4aa2f02b4024dd9584bd1db3c2595f61f1ecf7b985R1358, which I think may now be wrong) to consider that I don't want to creep into the scope of this PR... Would it be okay to leave this for a follow-up?

Reply:
That sounds fine.